challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java

```diff
@@ -0,0 +1,30 @@
+public class divyanshAg03{
+    static int SumOfSquares(int n)
+    {
+        if (n <= 3)    //base cases (ex:if n = 2 i.e. 1^2+1^2 = 2)
+        return n;
+
+        int res = n;
+
+        for (int i = 1; i<= n; i++)
+        {
+            int temp = i * i;
+            if (temp > n)
+            {
+                break;
+            }
+            else
+            {
+                res = Math.min(res, 1 + SumOfSquares(n - temp));
+            }
+        }
+        return res;
+    }
+
+    public static void main(String[] args) 
+    {   //running tests
+        System.out.println(SumOfSquares(8));
+        System.out.println(SumOfSquares(12));
+        System.out.println(SumOfSquares(9));
```

Comment:
You should pass the test cases presented in the original challenge. They're a _bit_ buried in the test file, but they go like this:

```
SumOfSquares(13)  -> 2
SumOfSquares(27)  -> 3
SumOfSquares(144) -> 1
SumOfSquares(84)   -> 3
SumOfSquares(85)   -> 2
```

Your solution passes the first two, but it hangs on the last three. It _looks_ like it's filling the stack up with recursive calls.

Reply:
I would probably recommend a second function of the form `bool canSum(int n, int[] numbers, int k)` that will take a number `n`, a list of numbers `numbers`, and an addend count `k` and check to see if any `k` numbers in `numbers` can sum to `n`. With that in hand, you should have no trouble checking to see if there's 2, 3, 4, etc. numbers that can sum to `n`.

Reply:
On it.

Reply:
Personally, I prefer to work on things on my own computer. Unless you deleted your branch, you can just make more commits against it. When you push the commits to your remote, they'll show up on the PR as well.